src/components/expression/expression-panel.js

```diff
@@ -113,8 +113,15 @@ genotet.ExpressionPanel.prototype.initPanel = function() {
     }.bind(this));
 
   // Input type update
+  var viewName = this.container.attr('id').replace('panel-view-', '');
```

Comment:
If you have the view object reference you can get its name by view.name(). It's in general not a good idea to first encoding some thing into a string and then decode it back from the string. Preferrably we should get the value directly.

Possible TODO.